frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -93,12 +93,17 @@ export function ViewTitleHeader(props) {
   const previousQuery = usePrevious(query);
 
   useEffect(() => {
-    if (!question.isStructured() || !previousQuestion?.isStructured()) {
+    const { isNative } = Lib.queryDisplayInfo(query);
+    const isPreviousQuestionNative =
+      previousQuery && Lib.queryDisplayInfo(previousQuery).isNative;
+
+    if (isNative || isPreviousQuestionNative) {
```

Comment:
Previously, when `previousQuestion` was `undefined`/`null`, `!previousQuestion?.isStructured()` evaluated to `true`. Now it evaluates to a falsy value. Not sure if it makes things better or worse, but to make it work exactly like before, we should do this:

```suggestion
    const isPreviousQuestionStructured =
      previousQuery && !Lib.queryDisplayInfo(previousQuery).isNative;

    if (isNative || !isPreviousQuestionStructured) {
```

Reply:
I found the original code before @kulyk rewrote it.
https://github.com/metabase/metabase/pull/18827/files#diff-7eae47ba62c31da65c448230eacac88be86470a204d450011acb631d7bb74144L80-L83

So I think we can safely assume that the intention was to make sure the both questions are structured.

Reply:
> Previously, when previousQuestion was undefined/null, !previousQuestion?.isStructured() evaluated to true. 

@kamilmielnik I'm 99% sure the older implementation (rewrite) made this assumption by mistake. Do you mind if I leave the code as is?